bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/ModuleUtil.java

```diff
@@ -0,0 +1,578 @@
+    }
+
+    /**
+     * Tries to convert raw value to its type.
+     *
+     * @param value raw value
+     * @param type the type to be converted to
+     * @return the converted value
+     * @throws Exception if conversion failed
+     */
+    private static Object convertToConfigType(String value, Type type) throws Exception {
+        Object result;
+        switch (type) {
+            case TEXT:
+                result = value;
+                break;
+            case BOOLEAN:
+                String trimmed = value.trim();
+                if (trimmed.equalsIgnoreCase("true")) {
+                    result = Boolean.TRUE;
+                } else if (trimmed.equalsIgnoreCase("false")) {
+                    result = Boolean.FALSE;
+                } else {
+                    result = null;
+                }
+                break;
+            case INTEGER:
+                result = Integer.valueOf(value);
+                break;
+            case DECIMAL:
+                result = Double.valueOf(value);
+                break;
+            default:
+                result = null;
+                break;
+        }
+        return result;
+    }
+
+    /**
+     * Validates if ConfigDescriptionParameter Type is equal to referred ConfigDescriptionParameter Type
+     *
+     * @param configDescription the source ConfigDescriptionParameter
+     * @param referredConfigName the referred ConfigDescriptionParameter name
+     * @param configDescriptionsMap Map with ConfigDescriptionParameter {@link #toConfigDescriptionMap(Set)}
+     */
+    private static void validateConfigType(ConfigDescriptionParameter configDescription, String referredConfigName,
+            Map<String, ConfigDescriptionParameter> configDescriptionsMap) {
+
+        Type configType = configDescription != null ? configDescription.getType() : null;
+        ConfigDescriptionParameter referredConfig = configDescriptionsMap.get(referredConfigName);
+        Type refferedConfigType = referredConfig != null ? referredConfig.getType() : null;
+        if ((configType == null || refferedConfigType == null) || !configType.equals(refferedConfigType)) {
+            throw new IllegalArgumentException("ConfigProperties Type missmatch:" + configDescription.getName() + ":"
+                    + configType + "->" + referredConfigName + ":" + refferedConfigType);
+        }
+    }
+
+    /**
+     * Validates source Input\Output value is assignableFrom Input\Output type.
+     *
+     * @param referredType the referred Input\Output type (fully qualified name i.e. java.lang.String)
+     * @param value the source Input\Output value
+     * @return true if is assignableFrom referred Input\Output type, false otherwise.
+     */
+    private static boolean validateConnectionType(String referredType, Object value) {
+        boolean valid = false;
+        try {
+            Class<?> clazzOutputExpected = FrameworkUtil.getBundle(ModuleUtil.class).loadClass((referredType));
```

Comment:
I don't have better idea how to cover such validation - but maybe it is redundant to compare again that the value matches the declared type of the Input. It could be enough to compare only the types of module Input with the Output of the other module. And at the end the module handler will check its input value when rule action is executed. If value is not matching the declared type it will throw exception at runtime. So better to remove the validation in this class above